play/src/components/joingame/EnterGameCode.tsx

```diff
@@ -20,17 +20,16 @@ const PaddedContainer = styled(Box)(({ theme }) => ({
 }));
 
 interface EnterGameCodeProps {
-  gameCodeValue: string;
-  setGameCodeValue: (newValue: string) => void;
+  handleGameCodeClick: (gameSessionId: string) => void;
```

Comment:
I rewrote this so that it returns a `bool` and so that it uses `async` and `await`. For my own benefit, is this better than using `Promise` `.then` like we were originally? I feel like this is cleaner now.

Reply:
In JS, they're kinda the same, the `async` method returns `Promise` based on [this doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function). Not sure how we previously were using the `Promise` while this method doesn't have any return value and not marked as `async`.